sdk/java-sdk-protobuf/src/main/scala/kalix/javasdk/impl/action/ActionsImpl.scala

```diff
@@ -380,4 +380,16 @@ class ActionContextImpl(override val metadata: Metadata, val messageCodec: Messa
   override def getGrpcClient[T](clientClass: Class[T], service: String): T =
     GrpcClients(system).getGrpcClient(clientClass, service)
 
+  override def componentGrpcClientMetadata: Metadata = {
+    if (metadata.has(Telemetry.TRACE_PARENT_KEY)) {
+      new MetadataImpl(
+        List(
+          MetadataEntry(
+            Telemetry.TRACE_PARENT_KEY,
+            MetadataEntry.Value.StringValue(metadata.get(Telemetry.TRACE_PARENT_KEY).get()))))
```

Comment:
Should we also propagate the `tracestate`?

Reply:
Interesting. I didn't add `tracestate` anywhere in our current tracing. I didn't think it was necessary to start with but I'm happy to change my mind. What will we/(our users) gain with it?

This is the description from [W3](https://www.w3.org/TR/trace-context/#tracestate-header)
```
The main purpose of the tracestate HTTP header is to provide additional vendor-specific trace identification information across different distributed tracing systems and is a companion header for the traceparent field. It also conveys information about the request’s position in multiple distributed tracing graphs.
```

Maybe to display graphs?

Reply:
Never mind. After our conversation now I know we should implement it per documentation https://www.w3.org/TR/trace-context/#design-overview

```
At a minimum they MUST propagate the traceparent and tracestate headers and guarantee traces are not broken. This behavior is also referred to as forwarding a trace.
```